skopt/space/space.py

```diff
@@ -721,3 +724,108 @@ def distance(self, point_a, point_b):
             distance += dim.distance(a, b)
 
         return distance
+
+    def get_dimension(self, id):
+        """Get the Dimension-object in the search-space using either
+        its name or index as the lookup-id.
+
+        Parameters
+        ----------
+        * `id` str or int:
+            If `id` is a string then it is the name of a dimension.
+            If `id` is an integer then it is the index of a dimension.
+
+        Returns
+        -------
+        * (`dimension`, `index`, `name`) [tuple(Dimension, int, str)]:
+            Dimension-object and its index in the search-space.
+            The name is taken from the Dimension-object if not None,
+            otherwise it is 'X_n' where n is the index.
+        """
+
+        if isinstance(id, str):
+            # The id is assumed to be a string-name of a dimension,
+            # so find its index.
+            index = self.dimension_names.index(id)
+        elif isinstance(id, int):
+            # The id is assumed to be the index of a dimension.
+            index = id
+        else:
+            raise ValueError("Dimension id must be either a string with the name or an integer for the index.")
+
+        # Get the corresponding dimension-object.
+        dimension = self.dimensions[index]
+
+        # Get the name of the dimension.
+        name = dimension.name
+
+        # Use a default name for unnamed dimensions.
+        # TODO: Maybe it would be better to set this in Space.__init__?
+        if name is None:
+            name = "X_" + str(index)
+
+        return dimension, index, name
+
+    def get_dimensions(self, ids=None):
```

Comment:
Could combine this with the above comment about using a `dict` like interface: `some_space[['lr', 'C']]` we would have to limit names to strings.

Reply:
What would be the right way of doing this in Python? Is there a dunder-method that can be overrided for this type of behaviour?

Reply:
This works now using `Space.__getitem__()` which can take either a single string or list of strings as argument.